Source/Diagnostics/WarpXOpenPMD.cpp

```diff
@@ -194,52 +255,90 @@ WarpXOpenPMDPlot::SavePlotFile (const std::unique_ptr<WarpXParticleContainer>& p
 
   openPMD::Iteration currIteration = m_Series->iterations[iteration];
   openPMD::ParticleSpecies currSpecies = currIteration.particles[name];
+  // meta data for ED-PIC extension
+  currSpecies.setAttribute( "particleShape", [](){
+      return double( WarpX::noz ); // TODO allow this per direction in the openPMD standard?
+  }() );
+  currSpecies.setAttribute( "particlePush", [](){
+      switch( WarpX::particle_pusher_algo ) {
+          case ParticlePusherAlgo::Boris : return "Boris";
+          case ParticlePusherAlgo::Vay : return "Vay";
+          case ParticlePusherAlgo::HigueraCary : return "HigueraCary";
+          default: return "other";
+      }
+  }() );
+  currSpecies.setAttribute( "particleInterpolation", [](){
+      switch( WarpX::field_gathering_algo ) {
+          case GatheringAlgo::EnergyConserving : return "energyConserving";
+          case GatheringAlgo::MomentumConserving : return "momentumConserving";
+          default: return "other";
+      }
+  }() );
+  currSpecies.setAttribute( "particleSmoothing", "none" );
+  currSpecies.setAttribute( "currentDeposition", [](){
+      switch( WarpX::current_deposition_algo ) {
+          case CurrentDepositionAlgo::Esirkepov : return "Esirkepov";
+          default: return "directBoris"; // TODO double-check if not "directMorseNielson" or "other"
```

Comment:
After offline checking the B&L again: We currently do `directMorseNielson` but `directBoris` might be interesting to explore for instability control as well.